msgpack/decode.ts

```diff
@@ -0,0 +1,244 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+import { EncodeType } from "./encode.ts";
+
+export function decode(uint8: Uint8Array) {
+  const pointer = { consumed: 0 };
+  const dataView = new DataView(uint8.buffer);
```

Comment:
I think we need to pass byteOffset and byteLength (`buffer` could be bigger than `uint8`)
```suggestion
  const dataView = new DataView(uint8.buffer, uint8.byteOffset, uint8.byteLength);
```

Reply:
Nice catch!